prow/jira/jira.go

```diff
@@ -190,6 +191,26 @@ func (jc *client) AddRemoteLink(id string, link *jira.RemoteLink) error {
 	return nil
 }
 
+func (jc *client) UpdateRemoteLink(id string, link *jira.RemoteLink) error {
+	internalLinkId := fmt.Sprint(link.ID)
+	req, err := jc.upstream.NewRequest("PUT", "rest/api/2/issue/"+id+"/remotelink/"+internalLinkId, link)
+	if err != nil {
+		return fmt.Errorf("failed to construct request: %w", err)
+	}
+	resp, err := jc.upstream.Do(req, nil)
+	if resp != nil {
+		defer resp.Body.Close()
+	}
+	if err != nil || resp.StatusCode != http.StatusNoContent {
```

Comment:
Can we just do this as:
```
if err != nil {
  return err
}
if resp.StatusCode != 204 {
  return fmt.Errorf("wrong status code")
}
```

That is much easier to follow than first allowing both cases and then subdividing between the two inside the `if ` statement